lib/msal-core/src/PopTokenGenerator.ts

```diff
@@ -0,0 +1,153 @@
+import { BrowserCrypto } from './BrowserCrypto';
+import { Base64 } from 'js-base64';
+import { HashAlgorithm, getStringFromArrayBuffer, KeyFormat, utf8Encode, getArrayBufferFromString, KeyUsages, KeyGenAlgorithm, getJwkString } from './PopTokenCommon';
+import { Utils } from './Utils';
+import { DatabaseCache } from './DatabaseCache';
+import { AuthenticationParameters } from './AuthenticationParameters';
+
+type CachedKeyPair = {
+    publicKey: CryptoKey,
+    privateKey: CryptoKey,
+    requestMethod: string,
+    requestUri: string
+}
+
+export class PopTokenGenerator {
+    private _crypto: BrowserCrypto;
+
+    private static DB_VERSION = 1;
+    private static DB_NAME = 'msal.db';
+    private static TABLE_NAME =`${PopTokenGenerator.DB_NAME}.keys`;
+    private _cache: DatabaseCache<CachedKeyPair>;
```

Comment:
maybe naming this cache to be more specific so it is not confused with our other caches

Reply:
Taking about `_cache` or `DatabaseCache`?